### ISSUE TITLE
Reorder may cause some error.

### ISSUE THREAD

**Issue by SunHowe** [Leopotam/ecslite-extendedfilters#1]
in EcsFilter source code:
```csharp
        internal void RemoveEntity (int entity) {
            if (AddDelayedOp (false, entity)) { return; }
            var idx = SparseEntities[entity] - 1;
            SparseEntities[entity] = 0;
            _entitiesCount--;
            if (idx < _entitiesCount) {
                _denseEntities[idx] = _denseEntities[_entitiesCount];
                SparseEntities[_denseEntities[idx]] = idx + 1;
            }
        }
```

SparseEntities[entityId] is equal to the index of entity in _denseEntities + 1.

But the _denseEntities element will be reorder in `Reorder`:
```csharp
        public static EcsFilter Reorder (this EcsFilter filter, EcsFilterReorderHandler cb) {
            var count = filter.GetEntitiesCount ();
            if (count > 1) {
                var entities = filter.GetRawEntities ();
                if (_filterSortPool.Length < entities.Length) {
                    Array.Resize (ref _filterSortPool, entities.Length);
                }
                for (int i = 0, iMax = count; i < iMax; i++) {
                    _filterSortPool[i] = cb (entities[i]);
                }
                Array.Sort (_filterSortPool, entities, 0, count);
            }
            return filter;
        }
```

**Comment by SunHowe** [comment]
`SparseEntities` should be reorder at this moment.

**Comment by Leopotam** [comment]
@SunHowe can you check `#develop` branch, is it works on your side? With `ecslite#develop` too for sure.

**Comment by SunHowe** [comment]
@Leopotam it works well at `develop` branch. Thanks you~

**Comment by Leopotam** [comment]
Thanks for feedback, will be released 22.09.2021.
